File: HackerRank Solutions/Algorithms/Strings/Strong Password.cpp
#include <bits/stdc++.h>

using namespace std;

// Complete the minimumNumber function below.
int minimumNumber(int n, string password) {
    int ans = 0;
    bool type[4] = {false};  
    for (int i=0; i<password.length(); i++) {
        if (password[i] >= '0' && password[i] <= '9') {
            type[0] = true;
        } else if (password[i] >= 'a' && password[i] <= 'z') {
            type[1] = true;
        } else if (password[i] >= 'A' && password[i] <= 'Z') {
            type[2] = true;
        } else {
            type[3] = true;
        }
    }
    for (int i=0; i<4; i++) {
        if (!type[i]) {
            ans++;
        }
    }    
    return max(ans, 6-n);
}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    int n;
    cin >> n;
    cin.ignore(numeric_limits<streamsize>::max(), '\n');

    string password;
    getline(cin, password);

    int answer = minimumNumber(n, password);

    fout << answer << "\n";

    fout.close();

    return 0;
}

File: HackerRank Solutions/Algorithms/Strings/Strong Password.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class Solution {

    // Complete the minimumNumber function below.
    static int minimumNumber(int n, String password) {
        // Return the minimum number of characters to make the password strong
        int ans = 0;
        if(!password.matches(".*[0-9].*")){
            ans++;
        }
        if(!password.matches(".*[a-z].*")){
            ans++;
        }
        if(!password.matches(".*[A-Z].*")){
            ans++;
        }
        if(!password.matches(".*[!@#$%^&*()+-].*")){
            ans++;
        }
        return Math.max(ans, 6-n);
    }

    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) throws IOException {
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        int n = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

        String password = scanner.nextLine();

        int answer = minimumNumber(n, password);

        bufferedWriter.write(String.valueOf(answer));
        bufferedWriter.newLine();

        bufferedWriter.close();

        scanner.close();
    }
}
